### DISCUSSION TITLE
Access response from a specific previous trial

### DISCUSSION THREAD

**Discussion by anjiecao** [jspsych/jsPsych#903] [Q&A] [answered]
Hello,

Apologize in advance if this is quite obvious from the documentation, but I am having some trouble understanding the sections on the data object. My ultimate goal is to create a branching timeline based on a participant's response, so: 
```
if the participant pressed button 0 on trial A1, then she will see trial A2, trial A3, trial A4. 
If the participant pressed button 1 on trial A1, then she will see trial B2, trial A3, trial B4. 
```
Basically, A1 and A3 are where I am collecting responses from the participants, whereas A2, A4, and B2, B4 are four different videos I am showing the participants in reaction to their responses at A1. 

Right now, I have already implemented the first part: 
```
if the participant pressed button 0 on trial A1, then she will see trial A2,
If the participant pressed button 1 on trial A1, then she will see trial B2,
```
by using [the instructions on conditional timeline ](https://www.jspsych.org/overview/timeline/#conditional-timelines). But when I am trying to extend my current solution to make the second branching, I encountered some issues to access the responses from trial A1. 

In the current conditional node, I have: 
```java
var data = jsPsych.data.get().last(1).values()[0]; // but I think this line should be adapted 
        if(data.key_press == jsPsych.pluginAPI.convertKeyCharacterToKeyCode('s')){
            return false;
        } else {
            return true;
        }
```
So my question is, how can I adapt the line accessing data so that I am not reading from the very last trial (which I assume is what `last(1)` is doing), but reading from the second-to-last trial?  I have also tried to use the function `var correct_trials = jsPsych.data.get().filter({trial_type: 'image-keyboard-response'});
`, but it still doesn't give me the correct response. I wonder if I am misunderstanding the way the data object is working. Any input is appreciated!

**Comment by vijaymarupudi** [comment, marked as answer]
Try changing it to `jsPsych.data.get().last(2).values()[0];`

**Reply by anjiecao** [reply]
Thank you! This is very helpful! But I wonder if there is a way to access the trial more flexibly. Because I am doing randomization, sometimes there would be like 3 or 4 trials before. Is there a way to calculate the distance between the current trial and the target trial and get the correct n for `last(n)`?

**Reply by anjiecao** [reply]
Never mind! i figured it out, combine filter w/ that line works: `var data = jsPsych.data.get().filter({trial_type : 'TYPE OF PREVIOUS NODE'}).values()[0];` Thank you so much!

**Reply by vijaymarupudi** [reply]
To make this more robust, I would add a data attribute to these trials, and filter on those.
